### PR TITLE
refactor: Cleanup of benchmark examples

### PR DESCRIPTION
**Pull request by mseeger**
Refactoring of 2 of our benchmark scripts, which makes it easier to understand them. I did this so I can show them in tutorials.

----

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: benchmarking/training_scripts/resnet_cifar10/resnet_cifar10.py [hunk 1]
```diff
@@ -178,7 +165,13 @@ def objective(config):
         sampler=valid_sampler,
         pin_memory=True,
     )
+    return train_loader, valid_loader
 
+
+def _create_training_objects(config):
+    num_gpus = config.get("num_gpus")
+    if num_gpus is None:
+        num_gpus = 1
```
Comment (wesk) [inline benchmarking/training_scripts/resnet_cifar10/resnet_cifar10.py:174]: What are the possible scenarios where `num_gpus is None`?

File: benchmarking/training_scripts/resnet_cifar10/resnet_cifar10.py [hunk 2]
```diff
@@ -189,33 +182,52 @@ def objective(config):
...
     )
     scheduler = torch.optim.lr_scheduler.MultiStepLR(
         optimizer, milestones=milestones, gamma=0.1
     )
+    return model, optimizer, scheduler
+
 
-    # Checkpointing
+def objective(config):
+    torch.manual_seed(np.random.randint(10000))
```
Comment (wesk) [inline benchmarking/training_scripts/resnet_cifar10/resnet_cifar10.py:197]: Should we set a manual seed here?

**Review by wesk — COMMENTED** [review]
Just two clarification questions

File: benchmarking/training_scripts/resnet_cifar10/resnet_cifar10.py (hunk 1 above)
Comment (mseeger) [reply benchmarking/training_scripts/resnet_cifar10/resnet_cifar10.py:174]: If the parameter is not given. We could deal with this in argparse, just saw that

File: benchmarking/training_scripts/resnet_cifar10/resnet_cifar10.py (hunk 2 above)
Comment (mseeger) [reply benchmarking/training_scripts/resnet_cifar10/resnet_cifar10.py:197]: This is not my code. I think Aaron wrote it.
We control seeds of everything on our side, but not what happens in the training functions.

Comment (mseeger) [reply benchmarking/training_scripts/resnet_cifar10/resnet_cifar10.py:197]: We definitely should not use the same seed, because that means that initial weights and batch ordering are the same for each evaluation, that would not be good.

Comment (mseeger) [reply benchmarking/training_scripts/resnet_cifar10/resnet_cifar10.py:197]: In the end, we cannot assume that we can control this for some user code, so let us not try to do it.

**Review by wesk — APPROVED** [review]
